plugins/azure/src/api/AzureSitesApiBackendClient.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AzureSitesApi } from './AzureSitesApi';
+import { SiteList } from '@backstage/plugin-azure-common';
+import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+
+/** @public */
+export class AzureSitesApiBackendClient implements AzureSitesApi {
+  private readonly identityApi: IdentityApi;
+  private readonly discoveryApi: DiscoveryApi;
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
+  }) {
+    this.discoveryApi = options.discoveryApi;
+    this.identityApi = options.identityApi;
+  }
+
+  async list({ name }: { name: string }): Promise<SiteList> {
+    try {
+      const url = `${await this.discoveryApi.getBaseUrl('azure')}/list/${name}`;
+      const { token: accessToken } = await this.identityApi.getCredentials();
+      const response = await fetch(url, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          ...(accessToken && { Authorization: `Bearer ${accessToken}` }),
+        },
+      });
+      return await response.json();
+    } catch (e: any) {
+      throw new Error('MissingAzureBackendException');
```

Comment:
Make sure the error is included in the message if you wish to wrap it up as it might be something else that's causing the error.